.github/workflows/publish.yml

```
@@ -44,3 +44,27 @@ jobs:
           push: true
           tags: ${{ steps.docker_metadata.outputs.tags }}
           labels: ${{ steps.docker_metadata.outputs.labels }}
+
+  deploy-k8s-staging:
+    runs-on: ubuntu-22.04
+    needs:
+      - publish-docker-images
```

```suggestion
      - publish-docker-images
      if: github.ref == 'refs/heads/main'
```

We can condition the deploy to main pushes only.